tools/clang/unittests/SPIRV/LibTestUtils.cpp

```diff
@@ -0,0 +1,228 @@
+        *targetEnv = SPV_ENV_VULKAN_1_0;
+      else if (targetEnvStr == "vulkan1.1")
+        *targetEnv = SPV_ENV_VULKAN_1_1;
+      else if (targetEnvStr == "vulkan1.1spirv1.4")
+        *targetEnv = SPV_ENV_VULKAN_1_1_SPIRV_1_4;
+      else if (targetEnvStr == "vulkan1.2")
+        *targetEnv = SPV_ENV_VULKAN_1_2;
+      else if (targetEnvStr == "vulkan1.3")
+        *targetEnv = SPV_ENV_UNIVERSAL_1_6;
+      else if (targetEnvStr == "universal1.5")
+        *targetEnv = SPV_ENV_UNIVERSAL_1_5;
+      else {
+        fprintf(stderr, "Error: Found unknown target environment.\n");
+        return false;
+      }
+      // Also push target environment to restArgs so that it gets passed to the
+      // compile command.
+      restArgs->push_back(tokens[i]);
+    } else {
+      restArgs->push_back(tokens[i]);
+    }
+  }
+
+  if (targetProfile->empty()) {
+    fprintf(stderr, "Error: Missing target profile argument (-T).\n");
+    return false;
+  }
+  // lib_6_* profile doesn't need an entryPoint
+  if (targetProfile->c_str()[0] != 'l' && entryPoint->empty()) {
+    fprintf(stderr, "Error: Missing entry point argument (-E).\n");
+    return false;
+  }
+  return true;
+}
+
+void convertIDxcBlobToUint32(const CComPtr<IDxcBlob> &blob,
+                             std::vector<uint32_t> *binaryWords) {
+  size_t num32BitWords = (blob->GetBufferSize() + 3) / 4;
+  std::string binaryStr((char *)blob->GetBufferPointer(),
+                        blob->GetBufferSize());
+  binaryStr.resize(num32BitWords * 4, 0);
+  binaryWords->resize(num32BitWords, 0);
+  memcpy(binaryWords->data(), binaryStr.data(), binaryStr.size());
+}
+
+bool compileCodeWithSpirvGeneration(const llvm::StringRef code,
+                                    const llvm::StringRef entryPoint,
+                                    const llvm::StringRef targetProfile,
+                                    const std::vector<std::string> &restArgs,
+                                    std::vector<uint32_t> *generatedBinary,
+                                    std::string *errorMessages) {
+  std::wstring srcFile(L"");
+  std::wstring entry(entryPoint.begin(), entryPoint.end());
+  std::wstring profile(targetProfile.begin(), targetProfile.end());
+
+  std::vector<std::wstring> rest;
+  for (const auto &arg : restArgs)
+    rest.emplace_back(arg.begin(), arg.end());
+
+  bool success = true;
+
+  try {
+    dxc::DxcDllSupport dllSupport;
+    IFT(dllSupport.Initialize());
+
+    if (hlsl::options::initHlslOptTable())
+      throw std::bad_alloc();
+
+    CComPtr<IDxcLibrary> pLibrary;
+    CComPtr<IDxcCompiler> pCompiler;
+    CComPtr<IDxcOperationResult> pResult;
+    CComPtr<IDxcBlobEncoding> pErrorBuffer;
+    CComPtr<IDxcBlob> pCompiledBlob;
+    CComPtr<IDxcIncludeHandler> pIncludeHandler;
+    HRESULT resultStatus;
+
+    bool requires_opt = false;
+    for (const auto &arg : rest)
+      if (arg == L"-O3" || arg == L"-O0" || arg.substr(0, 8) == L"-Oconfig")
```

Comment:
What about `-O1`, `-O2`?
Also, why is `-fcgl` added if no `-Ox` is specified? Wouldn't that change the default behavior we observe vs invoking the commandline (where `-O3` is the default?)
Why not forwarding those arguments directly?

Reply:
This was the existing behaviour. I was looking to change as little as possible from what was there before this code was removed.

Reply:
Again, I will keep the original behaviour. If we want to change it, it can be done in another PR. 

The reason was probably that we should default to the behaviour that we want most often, which is -fcgl for testing.